### DISCUSSION TITLE
No cache is expensive, can I invalidate static cache url where my cache tag is invalidated?

### DISCUSSION THREAD

**Discussion by tom-m-clarity-growth** [statamic/cms#10160] [Q&A] [answered]
I have tried using `{{ no_cache }}` around parts of my templates where content can change from time to time, but that has been damaging to my load times as really it is not truly dynamic content, e.g live data etc. Instead it is just a client who wants to update their navigation on a weekly basis to include upcoming event pages, or a blog index page that is only intermittently updated.

In the case of the navigation example thought about using:
```
{{ cache key="navigation }}
    {{ nav select="url|title" include_home="false" depth="3" }}
        ...
    {{ /nav }}
{{ /cache }}
```
with an event listener set up:
```
public function boot()
{
    parent::boot();

    // Main Navigation Invalidator
    Event::listen(function (NavTreeSaved $event) {
        if ($event->tree->handle() === 'pages') {
            Cache::forget('navigation');
        }
    });
}
```
but this is irrelevant if using static caching.

I thought about doing:
```
{{ nocache }}
    {{ cache key="navigation }}
        {{ nav select="url|title" include_home="false" depth="3" }}
            ...
        {{ /nav }}
    {{ /cache }}
{{ /nocache }}
```
but this is still not performant, and i'm questioning if logically it's even worth doing.

I was hoping there was a way I could programmatically find all URLs that have this cache tag and invalidate the URLs after the cache tag has been invalidated to ensure only pages affected by a relative update are invalidated and regenerated. This way I could have performant development and efficient static caching invalidation and regeneration. I realise I could create a custom invalidator for each of these tags, but what I'm struggling with is how to easily identify urls that would qualify because the cache tag with the relevant key is present. Not sure how to identify this.

Has anyone tried to achieve this before?

Or is there a much simpler solution I've completely missed?

**Comment by jasonvarga** [comment]
~You can check out https://github.com/tv2regionerne/statamic-cache. I believe that addon is doing almost exactly what you are describing. However I'm not sure of the status of it. It may still be in development.~

https://github.com/thoughtco/statamic-cache-tracker is what I was remembering.

**Reply by tom-m-clarity-growth** [reply]
Thanks for this, I'll take a look!

**Comment by ryanmitchell** [comment, marked as answer]
We have a similar add on - https://github.com/thoughtco/statamic-cache-tracker - it doesnt require a database so may be more suitable.

**Reply by tom-m-clarity-growth** [reply]
Thanks for this, I'll take a look!

**Reply by jasonvarga** [reply]
Actually this is the one I was thinking about in my other comment.
